### ISSUE TITLE
strcmp(): Passing null to parameter #1 ($string1) of type string is deprecated

### ISSUE THREAD

**Issue by fernando-rosa-polvo** [ADOdb/ADOdb#1021]
### Description
When using $myActiveRecord->Save() to UPDATE with NULL fields:

strcmp(): Passing null to parameter  #1 ($string1) of type string is deprecated
in strcmp called at /var/app/current/vendor/adodb/adodb-php/adodb-active-record.inc.php (1081)
in ADODB_Active_Record::Update called at /var/app/current/vendor/adodb/adodb-php/adodb-active-record.inc.php (848)
in ADODB_Active_Record::Save called at /var/app/current/src/...

### Environment
- ADOdb version: 5.22.6
- Database: MySQL 8.0 
- PHP version: 8.1
- Platform: Amazon Linux 2

**Comment by dregad** [comment]
Thanks for the bug report. 

> ADOdb version: 4.22.6

I assume that's a typo and you meant 5.22.6. Does the problem also exist in latest *master* branch ?

Could you please provide a code snippet to reproduce the problem ? Thank you

**Comment by fernando-rosa-polvo** [comment]
Hi.

> I assume that's a typo and you meant 5.22.6. Does the problem also exist in latest master branch ?

Yes, the problem exists int 5.22.6, 5.22.7 and master (235caae3cdf18f7714101f3a8d5a99a270b88201).

> Could you please provide a code snippet to reproduce the problem ?

This is the code to reproduce the problem:

```
<?php

// CREATE DATABASE test;
// CREATE TABLE person (id INT PRIMARY KEY, name VARCHAR(100));

error_reporting(E_ALL);

require_once "vendor/ADOdb/adodb.inc.php";
require_once "vendor/ADOdb/adodb-active-record.inc.php";

$connection = ADONewConnection("mysqli");
$connection->Connect("localhost", "root", "password", "test");
$connection->Execute("SET NAMES 'UTF8'");

ADODB_Active_Record::SetDatabaseAdapter($connection);

class Person extends ADODB_Active_Record {
    var $_table = 'person';
    var $id;
    var $name;
}

 // Create or update person for test
$person = new Person();
$person->Load("id = ?", [1]);
$person->id = 1;
$person->name = 'Jhon';
$person->Save();

 // Update name to null
$person = new Person();
$person->Load("id = ?", [1]);
$person->id = 1;
$person->name = null;
$person->Save(); // Deprecated: strcmp(): Passing null to parameter #1...
```

Thank you.

**Comment by dregad** [comment]
Sorry for the delayed response. 

I tried your snippet, and got this
```
$ php test.php
SQL Statement failed on preparation: SELECT * FROM PERSON WHERE id = ?'
SQL Statement failed on preparation: INSERT INTO PERSON (ID,NAME) VALUES (?,?)'
SQL Statement failed on preparation: SELECT * FROM PERSON WHERE id = ?'
SQL Statement failed on preparation: INSERT INTO PERSON (ID,NAME) VALUES (?,?)'
```

I was surprised at first, as the `person` table does exist in my test database. Then I realized that its name was somehow converted to uppercase, and [on Linux MySQL table names are case-sensitive](https://dev.mysql.com/doc/refman/8.0/en/identifier-case-sensitivity.html).

After further investigation, I found out the table name was converted to upper case by _adodb_quote_fieldname() - stack trace as of 1a2706136c00737e47bafcaf74b90338cf57402a:

```
adodb-lib.inc.php:696, _adodb_quote_fieldname()
adodb-active-record.inc.php:1141, ADODB_Active_Record->nameQuoter()
adodb-active-record.inc.php:801, ADODB_Active_Record->Load()
test.php:28, {main}()
```

With $ADODB_QUOTE_FIELDNAMES not specifically initialized, it defaults to `null`, ADOdb's behavior in this case is to convert to upper case. So I hit known bug #745. 

After setting `$ADODB_QUOTE_FIELDNAMES = 'NATIVE';`, your script works as expected, and I can confirm the behavior.
